llvm/docs/LangRef.rst

```diff
@@ -16464,8 +16464,8 @@ type.
 
 Semantics:
 """"""""""
-Follows the IEEE754 2008 semantics for minNum, except for handling of
-+0.0 vs -0.0. This matches the behavior of libm's fmin.
+Follows the IEEE754 2008 semantics for minNum.
+This also matches the behavior of libm's fmin.
```

Comment:
I actually think we should change the intrinsic to match the stricter 2019 signed zero behavior. The old behavior is equivalent to a stricter behavior with the nsz flag. 

This note about "libm's fmin" is too simple given the history of broken libms.

Reply:
Since C23 doesn't require it and libc implementions doesn't do it.
For some architectures, such as X86/ARM-preV8/MIPS-prer6, it will need more work to archive +0.0>-0.0.
So I think that we should have a intrinsic just suite for libc's fmin/fmax.

Of course, for architectures that have FMINNUM_IEEE/FMAXNUM_IEEE, we can map FMINNUM/FMAXNUM to the _IEEE  ones, just as we did for ARMv8 and MIPSr6. 

I will add more clairification about the history of libm.

Reply:
I you feel that the "IEEE754-2008 + +0.0>-0.0" is useful, we can export ISD::FMINNUM_IEEE as an new intrinsic.
Anyway, it will be another patch/PR.

Reply:
> For some architectures, such as X86/ARM-preV8/MIPS-prer6, it will need more work to archive +0.0>-0.0.

Correct, these will need some kind of lowering code.

> So I think that we should have a intrinsic just suite for libc's fmin/fmax.

No, you can implement these with an nsz flag on the one intrinsic with the stricter definition. You should end up with the same codegen if you know the sign of zero is insignificant. The last thing we need is more min/max intrinsic variants.

> we can export ISD::FMINNUM_IEEE as an new intrinsic.

There should be no new intrinsic. FMINNUM_IEEE should just be the minnum behavior for the existing intrinsic